extjsdk/src/main/java/io/vantiq/extjsdk/ExtensionWebSocketListener.java

```diff
@@ -314,6 +315,11 @@ public void onMessage(ResponseBody body) {
         else if (msg.get("op") == null) {
             log.debug("Http response received");
             if (client.isAuthed()) {
+                // Is an error message before successful connection to the target source
+                // This is most likely a failure related to a source connection request
+                if (!client.isConnected() && (Integer) msg.get("status") >= 300) {
+                    client.sourceFuture.complete(false);
```

Comment:
Probably worth logging a debug message here  to note that things failed.  Can be fixed next time around (that is, no reason to hold this up).